src/NovaVaultV2.sol

```diff
@@ -31,19 +50,38 @@ contract NovaVaultV2 {
         _;
     }
 
-    function addDex(address _contract) external {
+    modifier onlyAllowedDexContracts() {
```

Comment:
`onlyAllowedContracts` since we might use contract that are not DEX in the future, like the [PSM contract](https://github.com/makerdao/dss-psm)